flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/aggregate/MiniBatchGlobalGroupAggFunction.java

```diff
@@ -193,7 +193,7 @@ public void finishBundle(Map<RowData, RowData> buffer, Collector<RowData> out)
 
                 // if this was not the first row and we have to emit retractions
                 if (!firstRow) {
-                    if (!equaliser.equals(prevAggValue, newAggValue)) {
+                    if (stateRetentionTime > 0 || !equaliser.equals(prevAggValue, newAggValue)) {
```

Comment:
IMO, this logic should be ideally embedded in `globalAgg`. Since both `prevAggValue ` and `newAggValue ` are derived from `globalAgg` and since `globalAgg` has knowledge about ttl (`globalAgg.open(new PerKeyStateDataViewStore(ctx.getRuntimeContext(), ttlConfig))`) it should be better to have all the necessary logic in one place. WDYT?